eq-author/src/apollo/resolvers/index.js

```diff
@@ -1,73 +1,88 @@
 import gql from "graphql-tag";
+import { PAGES, ANSWERS } from "../../constants/validation-error-types";
 
 export const typeDefs = gql`
   extend type QuestionPage {
     isNew: Boolean!
   }
+
+  extend type BasicAnswer {
+    isNew: Boolean!
+  }
 `;
 
-const GET_NEW_PAGES_LIST = gql`
-  query newPagesList {
-    newPagesList @client
+const GET_NEW_ENTITY_LIST = gql`
+  query newEntityList {
+    newEntityList @client
   }
 `;
 
-const initialiseMultiplePages = (pages, cache) => {
-  const { newPagesList } = cache.readQuery({
-    query: GET_NEW_PAGES_LIST,
+const addNewEntity = (type, id, cache) => {
+  const { newEntityList } = cache.readQuery({
+    query: GET_NEW_ENTITY_LIST,
   });
+  newEntityList.push(`${type}_${id}`);
+  cache.writeData({ data: { newEntityList } });
+};
 
-  pages.forEach(questionPage => {
-    newPagesList.push(questionPage.id);
+const removeNewEntity = (type, id, cache) => {
+  const { newEntityList } = cache.readQuery({
+    query: GET_NEW_ENTITY_LIST,
   });
 
-  cache.writeData({ data: { newPagesList } });
+  const updatedPageList = newEntityList.filter(p => p !== `${type}_${id}`);
+  cache.writeData({ data: { newEntityList: updatedPageList } });
+};
+
+const initialiseMultiplePages = (pages, cache) => {
+  pages.forEach(questionPage => {
+    addNewEntity(PAGES, questionPage.id, cache);
+  });
 };
 
-const isNewQuestionPage = (questionPage, args, { cache }) => {
-  const { newPagesList } = cache.readQuery({
-    query: GET_NEW_PAGES_LIST,
+const isNewEntity = type => (questionPage, args, { cache }) => {
```

Comment:
Is it always a page here? If not it shouldn't be `questionPage`

Reply:
No, it was before this PR but I'd missed renaming the variable. Done now.